apps/challenges/views.py

```diff
@@ -2710,16 +2716,84 @@ def get_annotation_file_presigned_url(request, challenge_phase_pk):
             settings.MEDIAFILES_LOCATION, challenge_phase.test_annotation.name
         )
 
-    response = generate_presigned_url(
-        file_key_on_s3, challenge_phase.challenge.pk
+    response = generate_presigned_url_for_multipart_upload(
+        file_key_on_s3, challenge_phase.challenge.pk, num_file_chunks
     )
     if response.get("error"):
         response_data = response
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
-    response_data = {"presigned_url": response.get("presigned_url")}
+    response_data = {
+        "presigned_urls": response.get("presigned_urls"),
+        "upload_id": response.get("upload_id")
+    }
     return Response(response_data, status=status.HTTP_200_OK)
 
 
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def annotation_file_upload_complete(request, challenge_phase_pk):
+    """
+    API to complete multipart upload for a test annotation file
+
+    Arguments:
+        request {HttpRequest} -- The request object
+        challenge_phase_pk {int} -- Challenge phase primary key
+    Returns:
+         Response Object -- An object containing the presignd url, or an error message if some failure occurs
+    """
+    if settings.DEBUG or settings.TEST:
+        response_data = {
+            "error": "Sorry, this feature is not available in development or test environment."
+        }
+        return Response(response_data)
+    # Check if the challenge phase exists or not
+    try:
+        challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+    except ChallengePhase.DoesNotExist:
+        response_data = {"error": "Challenge Phase does not exist"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if not is_user_a_host_of_challenge(
+        request.user, challenge_phase.challenge.pk
+    ):
+        response_data = {
+            "error": "Sorry, you are not authorized for uploading an annotation file."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.data.get("parts") is None:
+        response_data = {"error": "Uploaded file Parts metadata is missing"}
```

Comment:
`Uploaded file Parts metadata is missing` -> `Uploaded file parts metadata is missing!`